mirror-index.sh

```diff
@@ -0,0 +1,107 @@
+            ;;
+            -v|--verbose)
+            set -x
+            shift 1
+            ;;
+            '--help')
+            print_usage
+            exit 0
+            ;;
+            *)
+            echo -e "Unknown option $1 is specified. See usage:\n"
+            print_usage
+            exit 0
+        esac
+    done
+}
+
+print_usage () {
+  echo "Usage:   $0 -i [ORIGINAL_INDEX] -t [TARGET_INDEX]"
+  echo "Options:
+  --index,  -i     the index which should be tested
+                   To find index see Index Image Location in cvp-test-report of
+                   web-terminal-operator-metadata-container-${WANTED_VERSION}
+                   http://external-ci-coldstorage.datahub.redhat.com/cvp/cvp-redhat-operator-bundle-image-validation-test/
+
+  --target, -t     the target where mirrored index should be pushed
+
+  --verbose, -v    print the verbose output during execution
+
+  --help,   -h     help
+  "
+}
+
+parse_arguments "$@"
+
+echo "Preparing mirrored index $INDEX_MIRROR for $INDEX_IMG"
+
+rm -rf $MIRROR_MANIFEST_FLD/
+
+# Go through $INDEX_IMG and grab all the mappings
+docker pull $INDEX_IMG
+
+oc adm catalog mirror --insecure=true --manifests-only --to-manifests=$MIRROR_MANIFEST_FLD $INDEX_IMG quay.io/wto
+
+cd $MIRROR_MANIFEST_FLD
+
+# Create the new mapped-images.txt that contains only the web-terminal mappings
+grep web-terminal mapping.txt > mapped-images.txt
+
+# cut chars after : to evaluate index repo
+INDEX_REPO=$(echo $INDEX_IMG | sed -e 's/^\(.*\):.*$/\1/')
+# Add the actual index into the mapping
+INDEX_DIGEST=$( docker inspect $INDEX_IMG | jq -r ".[].RepoDigests | .[] | select(contains(\"${INDEX_REPO}\"))")
+
+echo $INDEX_DIGEST=$INDEX_MIRROR >> mapped-images.txt
+
+docker tag $INDEX_IMG $INDEX_MIRROR
+docker push $INDEX_MIRROR
+
+# Fix the right side of mapped-images.txt. Replace images with their proper upstream counterparts
+sed -i 's/web-terminal-tech-preview-web-terminal-tooling-rhel8/web-terminal-tooling/g' mapped-images.txt
+sed -i 's/web-terminal-tech-preview-web-terminal-exec-rhel8/web-terminal-exec/g' mapped-images.txt
+sed -i 's/web-terminal-tech-preview-web-terminal-rhel8-operator/web-terminal-operator/g' mapped-images.txt
+sed -i 's/rh-osbs-web-terminal-operator-metadata/web-terminal-operator-metadata/g' mapped-images.txt
+
+# Fix the left side of mapped-images.txt. registry.redhat.io are not available until after the release happens so we defer to redhat proxy
+sed -i 's/registry.redhat.io\/web-terminal-tech-preview\/web-terminal-rhel8-operator/registry-proxy.engineering.redhat.com\/rh-osbs\/web-terminal-operator/g' mapped-images.txt
+sed -i 's/registry.redhat.io\/web-terminal-tech-preview\/web-terminal-exec-rhel8/registry-proxy.engineering.redhat.com\/rh-osbs\/web-terminal-exec/g' mapped-images.txt
+sed -i 's/registry.redhat.io\/web-terminal-tech-preview\/web-terminal-tooling-rhel8/registry-proxy.engineering.redhat.com\/rh-osbs\/web-terminal-tooling/g' mapped-images.txt
```

Comment:
Use `|` instead of `/` to avoid having to escape chars constantly.
```suggestion
sed -i 's|registry.redhat.io/web-terminal-tech-preview/web-terminal-rhel8-operator|registry-proxy.engineering.redhat.com/rh-osbs/web-terminal-operator|g' mapped-images.txt
sed -i 's|registry.redhat.io/web-terminal-tech-preview/web-terminal-exec-rhel8|registry-proxy.engineering.redhat.com/rh-osbs/web-terminal-exec|g' mapped-images.txt
sed -i 's|registry.redhat.io/web-terminal-tech-preview/web-terminal-tooling-rhel8|registry-proxy.engineering.redhat.com/rh-osbs/web-terminal-tooling|g' mapped-images.txt
```